### PR TITLE
feat: add dialogs.

### PR DESCRIPTION
**Pull request by marvinIsSacul**
## Release Notes

- **New Features**
	- Enhanced API client initialization with improved error handling for unauthorized access.
	- New functions for managing transaction reruns and bulk run filters.
	- Introduced new components for error messages, alerts, and confirmation dialogs.
	- Transitioned to a `DataGrid` for displaying transaction logs, improving performance and user experience.

- **Bug Fixes**
	- Improved error handling for HTTP requests and transaction fetching.

- **Refactor**
	- Updated several components for better state management and user interaction.
	- Enhanced type definitions for transactions and API responses.

- **Chores**
	- Updated dependencies and cleaned up unused code.

### CODE REVIEW

**Comment by drizzentic** [comment]
@coderabbitai review

File: packages/transaction-log/src/components/common/app.main.component.tsx [hunk 1]
```diff
@@ -315,20 +353,140 @@ const App: React.FC = () => {
...
 
+  const handleReRunMatches = async () => {
+    showBackdrop(<Loader />, true)
+
+
+    type BulkRunFilterCountParams = Parameters<typeof getBulkRunFilterCount>;
+
+    const params = {
+      ...getFilters() as any,
+      filterLimit: 1000,
```
Comment (drizzentic) [inline packages/transaction-log/src/components/common/app.main.component.tsx:375]: What happens in a situation we have more than 1000 matches? Has it been handled in the prompt?

File: packages/transaction-log/src/components/common/app.main.component.tsx [hunk 2]
```diff
@@ -316,19 +341,128 @@ const App: React.FC = () => {
...
     const searchTerm = searchQuery.toLowerCase()
     return [
+      // @ts-ignore
       transaction.channelName,
+      // @ts-ignore
       transaction.clientName,
       transaction.request.method,
       transaction.request.host,
       transaction.request.path,
+      // @ts-ignore
```
[the remark this answers is not in the document]
Comment (drizzentic) [reply packages/transaction-log/src/components/common/app.main.component.tsx:489]: @marvinIsSacul please address this as well.

File: packages/transaction-log/src/types/index.ts [hunk 3]
```diff
@@ -7,3 +7,73 @@ export type Channel = {
...
   name: string
 }
+
+export type TransactionStatus =
+  | 'Processing'
+  | 'Pending Async'
+  | 'Successful'
+  | 'Completed'
+  | 'Completed with error(s)'
+  | 'Failed'
```
[the remark this answers is not in the document]
Comment (drizzentic) [reply packages/transaction-log/src/types/index.ts:17]: is this suggestion worth considering @marvinIsSacul ?

**Review by drizzentic — CHANGES_REQUESTED** [review]

File: packages/transaction-log/src/components/common/app.main.component.tsx (hunk 1 above)
Comment (marvinIsSacul) [reply packages/transaction-log/src/components/common/app.main.component.tsx:375]: The legacy app uses 1000 as default.

**Review by drizzentic — APPROVED** [review]

**Pull request merged**